bindings/python/bipedal_locomotion_framework.cpp.in

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file bipedal_locomotion.cpp.in
+ * @authors Giulio Romualdi, Diego Ferigo
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <pybind11/pybind11.h>
+
+#include <BipedalLocomotion/bindings/Math/Module.h>
+#include <BipedalLocomotion/bindings/ParametersHandler/Module.h>
+
+@cmakeif FRAMEWORK_COMPILE_Contact
+#include <BipedalLocomotion/bindings/Contacts/Module.h>
+@endcmakeif FRAMEWORK_COMPILE_Contact
+
+@cmakeif FRAMEWORK_COMPILE_Planners
+#include <BipedalLocomotion/bindings/Planners/Module.h>
+@endcmakeif FRAMEWORK_COMPILE_Planners
+
+@cmakeif FRAMEWORK_COMPILE_RobotInterfaceBindings
+#include <BipedalLocomotion/bindings/RobotInterface/Module.h>
+@endcmakeif FRAMEWORK_COMPILE_RobotInterfaceBindings
+
+@cmakeif FRAMEWORK_USE_manif
+#include <BipedalLocomotion/bindings/manif/Module.h>
+@endcmakeif FRAMEWORK_USE_manif
+
+@cmakeif FRAMEWORK_COMPILE_FloatingBaseEstimators
+#include <BipedalLocomotion/bindings/FloatingBaseEstimators/Module.h>
+@endcmakeif FRAMEWORK_COMPILE_FloatingBaseEstimators
```

Comment:
In view of the possibility to import the bindings libraries downstream, maybe it could turn useful moving this logic to a public header that, when included, pulls all the symbols of wrapped classes.

Reply:
I don't know if it's really related to this comment however I've some doubts on how it is possible to import an existing binding in another project. Indeed the only way to expose symbols of the wrapped class to other bindings is by using `pybind11_add_module` with the option `SHARED`. However, I don't know if we can use this mechanism along with `PYBIND11_MODULE(bindings, m)`. 

I'm not actually able to see the entire path.

Reply:
Probably related https://github.com/pybind/pybind11/issues/1391

Reply:
I'm not sure either and this should be further tested. However, I think that a downstream project could do the following:

```cmake
find_package(BipedalLocomotion COMPONENTS Bindings)

pybind11_add_module(pybind11_downstream_module MODULE downstream_module.cpp)
target_link_libraries(pybind11_downstream_module PRIVATE BipedalLocomotion::TargetWithBindings OtherDownstreamLibs)
```

With `downstream_module.cpp`:

```cpp
#include <pybind11/pybind11.h>
#include <BipedalLocomotion/all_exported_bindings.h>  // This is the shared header file
#include "downstream/bindings.h"

// Create the Python module
PYBIND11_MODULE(bindings, m)
{
    namespace py = ::pybind11;
    
    // Here I'm not really sure. We need to make pybind aware of the blf wrapped types,
    // So I guess we have to instantiate a dummy blf module and add to it all the classes needed downstream

    // Then, classes from "downstream/bindings.h" potentially accepting objects from blf can be binded.
}
```